bitbot/reminder.go

```diff
@@ -0,0 +1,259 @@
+package bitbot
+
+import (
+	"fmt"
+	"github.com/whyrusleeping/hellabot"
+	log "gopkg.in/inconshreveable/log15.v2"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+)
+
+//nolint gochecknoglobals
+var (
+	location   *time.Location
+	timeFormat string
+)
+
+type eventType struct {
+	ID          int `gorm:"unique;AUTO_INCREMENT;PRIMARY_KEY"`
+	Channel     string
+	Author      string
+	Description string
+	People      string
+	Time        time.Time
+}
+
+var ReminderTrigger = NamedTrigger{ //nolint:gochecknoglobals,golint
+	ID:   "reminder",
+	Help: "Set up events and remind them to concerned people. Usage: !remind list|time|add|remove|join|part",
+	Condition: func(irc *hbot.Bot, m *hbot.Message) bool {
+		return m.Command == "PRIVMSG" && strings.HasPrefix(m.Trailing, "!remind")
+	},
+	Action: func(irc *hbot.Bot, m *hbot.Message) bool {
+		timeFormat = "2006-01-02 15:04"
+
+		var err error
+		location, err = time.LoadLocation("UTC")
+		if err != nil {
+			irc.Reply(m, "Something went wrong: Couldn't load timezone")
+			log.Error("Reminder : Couldn't load UTC timezone", err.Error())
+		}
+
+		b.DB.AutoMigrate(&eventType{})
+
+		splitMSG := strings.Split(m.Content, " ")
+		if len(splitMSG) < 2 {
+			irc.Reply(m, "Not enough arguments provided")
+			return true
+		}
+
+		switch splitMSG[1] {
+		case "time":
+			irc.Reply(m, getTime())
+		case "add":
+			irc.Reply(m, addeventType(m, irc))
+		case "remove":
+			irc.Reply(m, removeeventType(m))
+		case "list":
+			irc.Reply(m, listeventTypes(m, irc))
+		case "join":
+			irc.Reply(m, joineventType(m))
+		case "part":
+			irc.Reply(m, parteventType(m))
+		default:
+			irc.Reply(m, "Wrong argument")
+		}
+		return true
+	},
+}
+
+type wrongFormatError struct {
+	arg string
+}
+
+func (e *wrongFormatError) Error() string {
+	return fmt.Sprintf("%s : is not of the awaited !remind [command] [ID] format", e.arg)
+}
+
+func getMessageIDFromString(body string) (int, error) {
+	// Parse message
+	msg := strings.Split(body, " ")
+	isAnID, err := regexp.MatchString("[0-9]+", msg[2])
+	if err != nil {
+		log.Info("Not and ID :", err)
+	}
+
+	if len(msg) != 3 || !isAnID {
+		return -1, &wrongFormatError{body}
+	}
+	id, _ := strconv.Atoi(msg[2])
+	return id, nil
+}
+
+// Signal yourself as interested in an event (Facebook™)
+func joineventType(message *hbot.Message) string {
+	id, err := getMessageIDFromString(message.Content)
+	if err != nil {
+		return "Wrong command. format is : !remind join [ID]"
+	}
+
+	var event eventType
```

Comment:
Just a personal preference, but I like to take all declared vars like this and put them at the top of the function to make them easier to find